src/cascadia/TerminalSettingsEditor/Appearances.cpp

```diff
@@ -42,6 +43,143 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+                    wchar_t name[512];
+                    if (SUCCEEDED(names->GetString(localeIndex, name, length + 1)))
+                    {
+                        fontAxesTagsAndNames.Insert(tagString, winrt::hstring{ name });
+                        continue;
+                    }
+                }
+                // if there was no name found, it means the font does not actually support this axis
+                // don't insert anything into the vector in this case
+            }
+        }
+        return fontAxesTagsAndNames;
+    }
+
+    winrt::hstring Font::_axisTagToString(DWRITE_FONT_AXIS_TAG tag)
+    {
+        std::wstring result;
+        for (int i = 0; i < 4; ++i)
+        {
+            result.push_back((tag >> (i * 8)) & 0xFF);
+        }
+        return winrt::hstring{ result };
+    }
+
+    AxisKeyValuePair::AxisKeyValuePair(winrt::hstring axisKey, float axisValue, const Windows::Foundation::Collections::IMap<winrt::hstring, float>& baseMap, const Windows::Foundation::Collections::IMap<winrt::hstring, winrt::hstring>& tagToNameMap) :
+        _AxisKey{ axisKey },
+        _AxisValue{ axisValue },
+        _baseMap{ baseMap },
+        _tagToNameMap{ tagToNameMap }
+    {
+        if (_tagToNameMap.HasKey(_AxisKey))
+        {
+            int32_t i{ 0 };
+            // this loop assumes that every time we iterate through the map
+            // we get the same ordering
+            for (const auto tagAndName : _tagToNameMap)
+            {
+                if (tagAndName.Key() == _AxisKey)
+                {
+                    _AxisIndex = i;
+                    break;
+                }
+                ++i;
+            }
+        }
+    }
+
+    winrt::hstring AxisKeyValuePair::AxisKey()
+    {
+        return _AxisKey;
+    }
+
+    float AxisKeyValuePair::AxisValue()
+    {
+        return _AxisValue;
+    }
+
+    int32_t AxisKeyValuePair::AxisIndex()
+    {
+        return _AxisIndex;
+    }
+
+    void AxisKeyValuePair::AxisValue(float axisValue)
+    {
+        _baseMap.Remove(_AxisKey);
+        _AxisValue = axisValue;
+        _baseMap.Insert(_AxisKey, _AxisValue);
+        _PropertyChangedHandlers(*this, PropertyChangedEventArgs{ L"AxisValue" });
+    }
+
+    void AxisKeyValuePair::AxisKey(winrt::hstring axisKey)
+    {
+        _baseMap.Remove(_AxisKey);
+        _AxisKey = axisKey;
+        _baseMap.Insert(_AxisKey, _AxisValue);
+        _PropertyChangedHandlers(*this, PropertyChangedEventArgs{ L"AxisKey" });
+    }
+
+    void AxisKeyValuePair::AxisIndex(int32_t axisIndex)
+    {
+        _AxisIndex = axisIndex;
+
+        int32_t i{ 0 };
+        // same as in the constructor, this assumes that iterating through the map
+        // gives us the same order every time
+        for (const auto tagAndName : _tagToNameMap)
+        {
+            if (i == _AxisIndex)
+            {
+                AxisKey(tagAndName.Key());
+                break;
+            }
+            ++i;
+        }
```

Comment:
This is the manual index to key mapping we have to do because xaml struggles with combo box bindings